pandas/core/reshape/reshape.py

```diff
@@ -517,7 +517,7 @@ def factorize(index):
         # For homogeneous EAs, frame._values will coerce to object. So
         # we concatenate instead.
         dtypes = list(frame.dtypes._values)
-        dtype = dtypes[0]
+        dtype = dtypes[0] if len(dtypes) > 0 else object
```

Comment:
What if you return the Series right away if the dataframe is empty?

```
        if not frame.dtypes.empty:
            dtypes = list(frame.dtypes._values)
            dtype = dtypes[0]
        else:
            return Series([], dtype=object)
```

Reply:
But you would still need to solve unstacking from an empty series.

Reply:
likely you can just add a `not frame.empty and frame._is_homogenerous_type` on L516 might work